apps/src/templates/studioHomepages/TeacherHomepage.jsx

```diff
@@ -195,12 +196,15 @@ export const UnconnectedTeacherHomepage = ({
         <ProtectedStatefulDiv ref={flashes} />
         <ProtectedStatefulDiv ref={teacherReminders} />
         {showNpsSurvey && <NpsSurveyBlock />}
-        {specialAnnouncement && (
-          <MarketingAnnouncementBanner
-            announcement={specialAnnouncement}
-            marginBottom="30px"
-          />
-        )}
+        {specialAnnouncement &&
+          currentGlobalConfigurationFor(
+            'teacherHomepage.marketingAnnouncementBanner'
+          ) && (
```

Comment:
This function always only returns `false` when the component is marked `false` in the regional configuration. In this case `fa.yml` has:

```
teacher-homepage:
  marketing-announcement-banner: false
```

So, the key `teacherHomepage.marketingAnnouncementBanner` returns `false` and this removes the component from the page when the region is `fa` (so, when we go to `/global/fa/home` as a teacher)

Reply:
What if, instead of wrapping the component in configuration flags inside the parent component, we configure the components themselves based on the current page path?

### Global Region Configs
```javascript
const globalRegionConfigs = {
  header: {},
  footer: {},
  pages: {
    '.*/home.*': { // page path mask
      marketingAnnouncementBanner: false, // componentId config
    },
  },
};
```

### Global Region Wrapper
```javascript
const globalRegionWrapper = (WrappedComponent, componentId) => {
  // Retrieves the current page Global Region config based on the page path mask.
  const pageConfigKey = Object.keys(globalRegionConfigs.pages).find(
    pageMask => RexExp(pageMask).test(window.location.pathname)
  );
  const pageConfig = globalRegionConfigs.pages[pageConfigKey];

  // Renders the component if it is enabled for the page
  return !pageConfig || pageConfig[componentId]
    ? props => WrappedComponent({...props})
    : null;
};
```

### MarketingAnnouncementBanner.jsx
```javascript
const MarketingAnnouncementBanner = () => {
  ...
};

export default globalRegionWrapper(MarketingAnnouncementBanner, 'marketingAnnouncementBanner');
```

Reply:
This implementation is essentially not different from the configuration flags used previously:
https://github.com/code-dot-org/code-dot-org/blob/3fdcec4bffa0311bca35897e9ed6e126612c8e14/apps/src/templates/studioHomepages/TeacherHomepage.jsx#L200-L208 

My idea is to wrap the component once and import it already wrapped, allowing its configuration to be managed centrally from the config file for each page, instead of wrapping it separately on every page where we want to control its behavior.

Reply:
This implementation is essentially not different from the configuration flags used previously:
https://github.com/code-dot-org/code-dot-org/blob/3fdcec4bffa0311bca35897e9ed6e126612c8e14/apps/src/templates/studioHomepages/TeacherHomepage.jsx#L200-L208 

My idea is to wrap the component once and import it already wrapped, allowing its configuration to be managed centrally from the config file for each page, instead of wrapping it separately on every page where we want to control its rendering.